packages/server-wallet/src/models/objective.ts

```diff
@@ -177,6 +177,7 @@ export class ObjectiveModel extends Model {
     const objectiveIds = (
       await ObjectiveChannelModel.query(tx)
         .column('objectiveId')
+        .forUpdate()
```

Comment:
Two observations about this: 

1. You are only putting the lock on the join table `objectives_channels`
2. `forChannelIds` is not actually used in the objective manager.